### DISCUSSION TITLE
Set propagation=rshared for NEXTCLOUD_MOUNT

### DISCUSSION THREAD

**Discussion by chatziko** [nextcloud/all-in-one#2585] [Ideas] [answered]
Hello,

I was wondering, is there a way to set `propagation=rshared` (instead of `rprivate`) to the bind mount created by NEXTCLOUD_MOUNT? This would allow to recursively access nested bind mounts.

A good use-case is to be able to mount zfs snapshots to use with [Snapshots app](https://apps.nextcloud.com/apps/files_snapshots). Each snapshot is essentially a separate bind mount so `rshared` is needed to properly access them inside the container.

The [manual-install](https://github.com/nextcloud/all-in-one/tree/main/manual-install) would be an option, but I was hoping to still be able to use AIO.

Thanks in advance.

**Comment by szaimen** [comment]
Hi, we currently don't set this manually and just use dockers default:
https://github.com/nextcloud/all-in-one/blob/74b3a0d0dfa17e20fe239a69bdda5541c0fcc5b0/php/src/Docker/DockerActionManager.php#L217
https://github.com/nextcloud/all-in-one/blob/74b3a0d0dfa17e20fe239a69bdda5541c0fcc5b0/php/src/Docker/DockerActionManager.php#L230

In my test with an example vfatmount, mounts get indeed mounted recursively into the container as well. Maybe this is something that happens only with zfs snapshots?

**Reply by chatziko** [reply]
Thanks for the response, from what I can tell, nested directories that are __already__ mounted when the container is launched are accessible, but directories mounted later are not, unless I set `propagation=rshared` (or `rslave`).

**Comment by szaimen** [comment]
> from what I can tell, nested directories that are already mounted when the container is launched are accessible

I see. That is very unfortunate.

So what would be preferred then? rshared or rslave?

Also, I guess we would need to verify if this breaks on Docker Desktop:
![image](https://github.com/nextcloud/all-in-one/assets/42591237/376a9354-25d5-497d-b7a1-cab21af3eeb5)
From https://docs.docker.com/storage/bind-mounts/#configure-bind-propagation

**Reply by chatziko** [reply]
`rslave` should be enough for the snapshot use-case (haven't test it).

`rshared` is the most flexible, as it allows mounts inside the container to be visible outside. Not sure whether there is a good use-case for this in nextcloud, could be...

**Comment by szaimen** [comment]
So then we will go with rslave. Mounts from inside the containers do not need to be available from the outside and also should not be imho.

**Comment by chatziko** [comment]
Sounds great, I tested it and it works fine with zfs snapshots. Thanks a lot for fixing this!

**Comment by szaimen** [comment, marked as answer]
This will get addressed with https://github.com/nextcloud/all-in-one/pull/2601

**Reply by chatziko** [reply]
Looking forward to it, thanks again for the help!

**Reply by szaimen** [reply]
This is now released with v6.4.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Reply by chatziko** [reply]
I finally managed to test it in v7.0.0, it works great, many thanks!
